x-pack/plugins/security_solution/public/common/hooks/use_global_header_portal.tsx

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState } from 'react';
+import { createPortalNode } from 'react-reverse-portal';
```

Comment:
can we use `EuiPortal` instead?

Reply:
`EuiPortal` doesn’t appear to have an equivalent of `react-reverse-portal`’s `createPortalNode`, which is the glue that allows the `FiltersGlobal` component in `x-pack/plugins/security_solution/public/common/components/filters_global/filters_global.tsx` to specify the same `InPortal` node that the `HeaderGlobal` component refers to in the `OutPortal`, in `x-pack/plugins/security_solution/public/common/components/header_global/index.tsx`.

I think we're using `react-reverse-portal` for a similar use cases, where one component is rendering to an `InPortal`, and another component is rendering an `OutPortal`, and they are tied-together via a single, common node. `react-reverse-portal` also tries to avoid re-renders when re-parenting components, FWIW.